common/CMakeLists.txt

```diff
@@ -1,12 +1,19 @@
-add_library(common_json STATIC json/impl/nlohmann/nlohmann_json.cpp)
-target_include_directories(common_json PRIVATE ../vendor/json/include/)
+set(json_sources "$<$<STREQUAL:${PLATFORM},linux_x86>:nlohmann/nlohmann_json.cpp>")
+set(kv_db_sources "$<$<STREQUAL:${PLATFORM},linux_x86>:lmdb/lmdb_kv_db.cpp>")
+
+add_library(common_json STATIC json/impl/${json_sources})
 target_include_directories(common_json PRIVATE ../)
+if(${json_sources} MATCHES ".*nlohman.*")
```

Comment:
Nitpick: nlohmann has two n's.

Reply:
Oh, right! This has bitten my ass multiple times already.